.github/workflows/contrib-openai.yml

```diff
@@ -37,19 +37,38 @@ jobs:
           pip install -e .
           python -c "import autogen"
           pip install coverage pytest-asyncio
-      - name: Install packages for test when needed
+      - name: Install packages for Retrievechat
         run: |
           pip install docker
           pip install qdrant_client[fastembed]
           pip install -e .[retrievechat]
-      - name: Coverage
+      - name: Coverage Retrievechat
         env:
           OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
           AZURE_OPENAI_API_KEY: ${{ secrets.AZURE_OPENAI_API_KEY }}
           AZURE_OPENAI_API_BASE: ${{ secrets.AZURE_OPENAI_API_BASE }}
           OAI_CONFIG_LIST: ${{ secrets.OAI_CONFIG_LIST }}
         run: |
           coverage run -a -m pytest test/agentchat/contrib/test_retrievechat.py test/agentchat/contrib/test_qdrant_retrievechat.py
+
+      - name: Install packages for MathChat
+        id: install_mathchat
+        if: ${{ always() }}
+        run: |
+          pip install docker
+          pip install -e .[mathchat]
+      - name: Coverage MathChat
+        if: ${{ always() && steps.install_mathchat.outcome == 'success' }}
+        env:
+          OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+          AZURE_OPENAI_API_KEY: ${{ secrets.AZURE_OPENAI_API_KEY }}
+          AZURE_OPENAI_API_BASE: ${{ secrets.AZURE_OPENAI_API_BASE }}
+          OAI_CONFIG_LIST: ${{ secrets.OAI_CONFIG_LIST }}
+        run: |
+          coverage run -a -m pytest test/agentchat/contrib/test_math_user_proxy_agent.py
```

Comment:
Looks like the output of coverage run will overwrite the output of the former step. In the end, only one step of coverage will be uploaded.

Reply:
Could we add different names to the output of different steps and combine them at  the upload step?

Reply:
The `coverage run -a` will append outputs to the same file(`.coverage`), so we are good with this. Not very sure but this is what I find.